topy/topy.py

```diff
@@ -121,11 +121,27 @@ def sanitize_filename(filename):
 def print_diff(filename, old, new, stream=sys.stdout):
     """Diffs the `old` and `new` strings and prints as unified diff to file-like object `stream`."""
 
-    # TODO: color output for terminals
     lines = unified_diff(old.splitlines(True), new.splitlines(True), filename, filename)
+    lines = map(lambda l: determine_color(l, stream=stream), lines)
     stream.writelines(lines)
 
 
+def determine_color(line, stream):
+    """Determines the color of the diffs."""
+    return add_output_color(line) if is_color_output_required(stream) else line
+
+
+def is_color_output_required(stream):
+    """Determines whether to output the line diffs in color or not."""
+    return False if opts.color == "never" else True if opts.color == "always" else hasattr(stream, 'isatty') and stream.isatty()
+
+
+def add_output_color(line):
+    """Adds color to the output of the diff lines."""
+    return f'\033[1;32m{line}\033[0;32m' if line.startswith('+') else (
+        f'\033[1;31m{line}\033[0;31m' if line.startswith('-') else f'\033[0m{line}')
```

Comment:
The color sequence is only needed at the *beginning* of each line, it's not needed at the end.

Instead you could end every line with the reset sequence (`\033[0m`), then uncolored lines don't need to be changed at all.

Reply:
Note that the f-string syntax is not yet available in Python 3.5, that's why the tests failed: https://github.com/intgr/topy/pull/25/checks?check_run_id=1807889579

Since Python 3.5 is end-of-life (https://www.python.org/downloads/release/python-3510/), we should just drop 3.5 support. There are three places that need changing: `setup.py`, `.travis.yml` and `README.rst`

Reply:
Okay, do you want me to edit the setup.py, .travis.yml and README.rst, or will you do them in a separate PR?

Reply:
Sorry, I was late replying to this. Doing it in this PR is fine :)

Reply:
No worries - I've made those changes - let me know if I've missed removing python3.5 from anywhere